server.js

```diff
@@ -14,10 +17,60 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+let messages = [welcomeMessage];
+
+app.get("/", function (req, res) {
+  res.sendFile(__dirname + "/index.html");
+  // res.json("I am here!");
+});
+
+// Search text in messages -  /messages/search?text=express
+app.get("/messages/search", (req, res) => {
+  const {text }= req.query;
+  const filteredMessages = messages.filter((message) =>
+   message.text.includes(text));
+  res.json(filteredMessages);
+});
+
+// get last 10 messages -  /messages/latest
+app.get("/messages/latest", (req, res) => {
+  res.json(messages.slice(-10));
```

Comment:
Great job getting the last 10 messages in the array. You could also sort by ID before slicing, to ensure they are actually the latest.